lox/main.cpp

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) assignUser
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <fmt/core.h>
+#include <fmt/format.h>
+
+#include <fstream>
+#include <lyra/arg.hpp>
+#include <lyra/help.hpp>
+#include <lyra/lyra.hpp>
+#include <numeric>
+#include <optional>
+#include <tl/expected.hpp>
+#include <tl/optional.hpp>
+#include <variant>
+#include <vector>
+
+template <typename stream>
+std::vector<std::string> read_input(stream& input) {
```

Comment:
ifstream inherits from istream so you don't need a template
```suggestion
std::vector<std::string> read_input(std::istream& input) {
```

Reply:
Ah I tried that but used the wrong class ^^